source/GameInterface/Services/Equipments/Patches/EquipmentTypeFieldPatch.cs

```diff
@@ -0,0 +1,76 @@
+using Common.Logging;
+using Common.Messaging;
+using GameInterface.Policies;
+using GameInterface.Services.Equipments.Messages.Events;
+using GameInterface.Services.ObjectManager;
+using HarmonyLib;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using TaleWorlds.Core;
+
+/*
+namespace GameInterface.Services.Equipments.Patches;
```

Comment:
A manual field sync with a transpiler for initializing the `_itemSlots` is required sadly

Reply:
```suggestion
[HarmonyPatch(typeof(Equipment))]
internal class EquipmentFieldPatches
{
    private static IEnumerable<MethodBase> TargetMethods() => AccessTools.GetDeclaredMethods(typeof(Equipment));



    [HarmonyTranspiler]
    private static IEnumerable<CodeInstruction> UpdateClanSettlementAutoRecruitment(IEnumerable<CodeInstruction> instructions)
    {
        var itemSlots = AccessTools.Field(typeof(Equipment), nameof(Equipment._itemSlots));

        foreach (var instruction in instructions)
        {
            // When storing the field _itemSlots
            if (instruction.opcode == OpCodes.Stfld &&
                instruction.operand as FieldInfo == itemSlots)
            {
                yield return instruction;

                // This adds a call after when _itemSlots field is set for all methods in the Equipment class
                yield return new CodeInstruction(OpCodes.Ldarg_0); // loads the Equipment instance onto the stack
                yield return new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(EquipmentFieldPatches), nameof(ItemSlotsCreatedIntercept)));
                
            }
            else
            {
                yield return instruction;
            }
        }
    }

    public static void ItemSlotsCreatedIntercept(Equipment instance)
    {
        // Publish a message here (should only contain the instance)
        // Handle in handler
        // Send to all clients
        // Set _itemSlots to "new EquipmentElement[12]" manually on the network handler
    }
}
```

Reply:
For the latest push I was able to avoid the ItemSlots Intercept. Please review.
In case it is needed I can get it back, but due to the nature how Equipment(Equipment) ctor is definded, with a loop setting all the itemSlots elements individually I think it is not the best approach.